src/valhalla_associate_segments.cc

```diff
@@ -69,58 +71,37 @@ enum class MatchType : uint8_t {
   kShortestPath = 1
 };
 
-struct MatchStats {
-  uint32_t success_count;
-  uint32_t failure_count;
-  uint32_t walk_count;
-  uint32_t path_count;
-
-  MatchStats()
-    : success_count(0),
-      failure_count(0),
-      walk_count(0),
-      path_count(0) {
+// Matched edge
+struct EdgeMatch {
+  GraphId edgeid;
+  uint32_t length;
+  float start_pct;
+  float end_pct;
+
+  EdgeMatch(const GraphId& e, const uint32_t l, const float start, const float end)
```

Comment:
theres no need for a constructor in a POD like this. you can always make it by just doing `EdgeMatch{e, l, st, en}` and since this constructor does nothing its just extra code for no value